core/modfunc.go

```diff
@@ -137,6 +137,13 @@ func (fn *ModuleFunction) setCallInputs(ctx context.Context, opts *CallOpts) ([]
 			return nil, fmt.Errorf("failed to convert arg %q: %w", input.Name, err)
 		}
 
+		if len(arg.metadata.Ignore) > 0 {
+			converted, err = fn.applyIgnoreOnDir(ctx, opts.Server, arg.metadata, converted)
+			if err != nil {
+				return nil, fmt.Errorf("failed to apply ignore pattern on arg %q: %w", input.Name, err)
+			}
+		}
```

Comment:
For a directory provided via CLI, and uploaded via `bk.LocalImport`, this will filter the resulting directory another time, but with this, right?
```graphql
{
    directory { 
        withDirectory(path: "/", directory: "<ID>", exclude: ["<same ignore pattterns>"]) 
    }
}
```

I'd rather find a fix so it doesn't filter twice, before enabling this for all directories. Simply having support for the CLI is enough to deprecate _views_.

This is what's causing the issue with `sdk/python/dev` by the way.

Reply:
> For a directory provided via CLI, and uploaded via bk.LocalImport, this will filter the resulting directory another time, but with this, right?

You're right, so you suggest that for now, we only applied //+ignore when the module is called from the CLI?

Reply:
Yes.

Reply:
Looks like it fixed the issue! I'm trying to get the CI green and then we can do the ignore process in a follow up

Reply:
Hm, I disagree here - I think the filtering twice is frustrating from a performance point of view, but I'd rather that than have entirely different behavior when invoking from the CLI vs from another module.

Is your concern just performance @helderco? Or is there a correctness concern here as well?

---

As a suggestion, the way to track this would be add a field to `Directory`s that indicate which `ignore` patterns were used during `resolveDirectoryFromCaller`. Then we just check if those were equal to the patterns specified here, and if they are, we can skip the `applyIgnoreOnDir`.

This should be pretty simple, it just needs to be plumbed through?